src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs

```diff
@@ -71,8 +71,15 @@ public bool Equals(EdmTypeCacheItem x, EdmTypeCacheItem y)
 
             public int GetHashCode(EdmTypeCacheItem obj)
             {
-                string combined = $"{obj.EdmType.FullTypeName()}~{obj.Nullable}";
-                return combined.GetHashCode();
+                unchecked
+                {
+                    int hashCode = 17;
+                    hashCode = (hashCode * 31) + obj.EdmType.FullTypeName().GetHashCode(); // Ideally we'd avoid FullTypeName since it
```

Comment:
Or can we trust it? the `Equals` implementation assumes that `Equals` is implemented, which it isn't, so that would suggest we're dealing with instance equality...

Reply:
I've convinced myself that the default `GetHashCode` implementation will work for us.